Alternate the method to elevate user privilege in REX (#2446)
theforeman/foreman-documentation: guides/common/modules/proc_altering-the-privilege-elevation-method.adoc

[id="altering-the-privilege-elevation-method_{context}"]
= Altering the Privilege Elevation Method

By default, push-based remote execution uses `sudo` to switch from the SSH user to the effective user that executes the script.
In some situations, you may require to use another method, such as `su` or `dzdo`.

> You can globally configure an alternative method to switch the effective user in your {Project} settings.

Reviewer: Maybe mention that "sudo" is the default value?

Author: I was trying to do that in the first sentence. Isn't that clear enough?

Reviewer: Haha, I've missed that :joy:

.Prerequisite

> 

Reviewer: ```suggestion
```

> * If you want to use `dzdo` for Ansible jobs, you have to install the `community.general` Ansible collection, which contains the *dzdo* Ansible plugin.

Reviewer: ```suggestion
* If you want to use `dzdo` for Ansible jobs, you have to install the `community.general` Ansible collection, which contains the *dzdo* Ansible plug-in.
```

ifdef::satellite[]
For more information, see https://access.redhat.com/solutions/7025758[Running Ansible job on hosts registered with Red Hat Satellite 6 fails with error could not find matching plugin: 'dzdo'] in the _Red{nbsp}Hat Knowledgebase_.
endif::[]

.Procedure

> 

Reviewer: ```suggestion
```

> . Navigate to *Administer* > *Settings* > *Remote Execution*.

Reviewer: I am unsure about "X > Y > Z" vs "X > Y" + "On the Z tab, click ...". I believe we currently have both versions within the docs. Any general thoughts? @adamlazik1 @asteflova 

For this instance: your call Lena.

Author: I've decided to add the tab selection as an additional but separate step.

Reviewer 2: > I am unsure about "X > Y > Z" vs "X > Y" + "On the Z tab, click ...". I believe we currently have both versions within the docs. Any general thoughts? @adamlazik1 @asteflova

A bit off-topic (let me know if you want to talk about this elsewhere), but I've been wondering lately: To what extent do you as a team strive to be consistent in these things? Is it expected to stick to the same wording across all content? Or just within a single guide? Within an assembly? Within a module?

Reviewer: Due to the modularity, I strive for consistency for everything within `guides/common/modules/`. If there is already some inconsistency (like 30:70) in the docs, then I don't strongly recommend going with the majority. If it's more like 5:95, then I always recommend to go with the 95%. I'd rather handle such cases in separate PRs.

Always remember: the docs is what we make it; conventions and consistency is human-made by us. :v:

[id="altering-the-privilege-elevation-method_{context}"]
= Altering the Privilege Elevation Method

By default, push-based remote execution uses `sudo` to switch from the SSH user to the effective user that executes the script on your host.

> In some situations, you may require to use another method, such as `su` or `dzdo`.

Reviewer 2: ```suggestion
In some situations, you might require to use another method, such as `su` or `dzdo`.
```

> You can globally configure an alternative method to switch to the effective user in your {Project} settings.

Reviewer 2: ```suggestion
You can globally configure an alternative method to switch to the effective user in your {Project} settings, such as `su` or `dzdo`.
```
With this, you could actually remove line 5.

Author: I'd prefer it with the earlier sentence.

.Prerequisite

> * If you want to use `dzdo` for Ansible jobs, you have to install the `community.general` Ansible collection, which contains the *dzdo* Ansible plug-in.

Reviewer 2: ```suggestion
* Ensure the Ansible collection that provides the required plug-in is installed.
For example, to use `dzdo`, ensure the `community.general` collection is installed.
```

Author: It's meant to be just for dzdo, because the other methods that Satellite provides (su, sudo), are part of ansible-core which is included by default and doesn't need an additional Ansible collection.

Reviewer: I'd go with "module" instead of "plug-in"; unless you want to say "'become' plug-in": https://docs.ansible.com/ansible/latest/collections/community/general/index.html#plugin-index

Other than that, I like Anet's suggestions.

Author: @maximiliankolb @asteflova I've updated what I agreed with.

ifdef::satellite[]

> For more information, see https://access.redhat.com/solutions/7025758[Running Ansible job on hosts registered with Red Hat Satellite 6 fails with error could not find matching plugin: 'dzdo'] in the _Red{nbsp}Hat Knowledgebase_.

Reviewer 2: I don't think you should link to the solution. If a user lands on this module, they will not encounter the error described in the solution after following your procedure.

If anything, I believe the solution should link to your new module.

Author: Probably just the Ansible collection installation link will suffice.

endif::[]
ifndef::satellite[]

> For more information, see https://docs.ansible.com/ansible/latest/collections_guide/collections_installing.html[Installing collections] in _Ansible documentation_.

Reviewer 2: This is useful for Satellite as well, so I suggest removing the ifndef.
